roles/openshift_node/tasks/registry_auth.yml

```diff
@@ -30,13 +30,34 @@
   delay: 5
   until: node_oreg_auth_credentials_create is succeeded
 
+- name: Create credentials for any additional registries
+  docker_creds:
+    path: "{{ oreg_auth_credentials_path }}"
+    registry: "{{ item.host }}"
+    username: "{{ item.user | default('openshift') }}"
+    password: "{{ item.password }}"
+    # Test that we can actually connect with provided info
+    test_login: "{{ item.test_login | default(omit) }}"
+    proxy_vars: "{{ l_docker_creds_proxy_vars }}"
+    image_name: "{{ item.image_name | default('openshift3/ose-pod') }}"
+    tls_verify: "{{ item.tls_verify | default(omit) }}"
+  when:
+  - openshift_additional_registry_credentials is defined
+  register: node_additional_registry_creds
+  retries: 3
+  delay: 5
+  until: node_additional_registry_creds is succeeded
+  with_items:
+    - "{{ openshift_additional_registry_credentials | default({}) }}"
+
 # Container images may need the registry credentials
 - name: Setup ro mount of /root/.docker for containerized hosts
   set_fact:
     l_bind_docker_reg_auth: True
   when:
     - openshift_is_atomic | bool
-    - oreg_auth_user is defined
+    - oreg_auth_user is defined or openshift_additional_registry_credentials is defined
```

Comment:
I think this would be better as
```yaml
- openshift_is_atomic | bool
- oreg_auth_user is defined or openshift_additional_registry_credentials is defined or node_oreg_auth_credentials_stat.stat.exists
```

Logic:  If we defined oreg_auth_user or openshift_additional_registry_credentials, we know for certain we will be updating/creating the .docker/config.json file.  If node_oreg_auth_credentials_stat.stat.exists, then it is already on disk (someone created that outside of openshift-ansible, etc), we don't need to check for changed because changed will be covered by defining the auth variables.

Reply:
ok, updated